src/components/mission.js

```diff
@@ -0,0 +1,46 @@
+import { React, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { retrieveInfo, updateState } from '../redux/mission/mission';
+
+const MissionComponent = () => {
+  const dispatch = useDispatch();
+  const missionData = useSelector((state) => state.missionReducer);
+  useEffect(() => {
+    dispatch(retrieveInfo([]));
+  }, []);
+
+  const handleMissionStateClick = (key, status) => {
+    if (status) {
+      dispatch(updateState({
+        id: key,
+        value: false,
+      }));
+    } else {
+      dispatch(updateState({
+        id: key,
+        value: true,
+      }));
+    }
```

Comment:
- [ ] All logic should be in the reducers, then dispatched on react view files. Please, the status change and this kind of thing is something you could add on the reducers, please work on it. 🐶